mlrun/datastore/targets.py

```diff
@@ -121,14 +123,23 @@ def validate_target_paths_for_engine(targets, engine, source):
                     f"spark CSV/Parquet targets must be directories, got path:'{target.path}'"
                 )
             elif engine == "pandas":
-                if source.attributes.get("chunksize"):
+                if isinstance(source, DataSource) and source.attributes.get(
```

Comment:
Please add a comment explaining why this check is needed - it's not self-apparent that the `source` may be a Pandas `DataFrame` (and the code does nothing to help with that), so at least comment about it. 
Even better - note that in the docstring of the function. 
Best - change the function signature to have a type hint - `source: Union[DataSource, pd.DataFrame]` (or something like that).